[3주차] 송은수 미션 제출합니다.

songess opened the pull request:
# 배포링크
[카카오톡](https://ceos-week3-react-messenger.vercel.app/)

# 후기 및 구현기능

생각보다 할 게 많았던 과제였던거 같습니다..!

디자이너분과 협업을 하면서 진행했기 떄문에 `styled-components`의 `theme`을 사용해 `font`, `textStyles`을 설정하고 시작했습니다. 또 `375x812` 맞춤으로 코드를 작성할 것이라 `rem`단위를 사용하지 않고 `px`을 사용했습니다.

절대경로를 설정했는데, cra를 사용한 프로젝트이기 때문에 `baseUrl`및 `paths`만으로는 설정이 부족했고 `craco`, `react-app-alias`의 라이브러리의 도움을 받아 해결했습니다. 다만 `@types`는 적용이 되지 않아 `type`들은 직접 선언해서 사용했습니다.

과제에 `recoil`사용이 권장되어 있어서 사용해봤습니다. `recoil`을 사용해보지 않았던 이유가 `redux`처럼 보일러 플레이트가 많고, 디스패치를 많이 사용해야 할 것 같아 사용해보지 않았으나, 직접 사용해보니 잘못된 생각이였다는 걸 깨달았습니다. `useRecoilState`를 `useState`처럼 사용할 수 있었고, 보일러 플레이트도 필요하지 않았습니다. 또한 `selector`를 통해 디스패치의 기능을 훨씬 쉽게 사용한 느낌이였습니다.  
공식문서를 찾아보니 `effects`를 통해 다른 `built-in`함수들도 많아 찾아보던 중 `onSet`과 `setSelf`는 유용하게 사용할 수 있을 거 같아 적용시켜 보았습니다.

`effects`는 '부수효과를 관리하고 atom을 초기화 또는 동기화하기 위한 API'라고 나와있어 `localStorage`에 제격이라 생각했습니다.
`setSelf`는 초기값을 설정하는 함수로, `localStorage`에 값이 있다면 받아오고, 없다면 저장해놓은 `json`데이터를 가져왔습니다.  
`onSet`은 atom을 모니터하며 값이 갱신되면 함수가 실행되어 `localStorage`를 최신화 시켰습니다.

이름을 눌렀을때 변하는 것을 구현하기 위해 `recoil`을 사용해 현재`user`가 누구인지를 토클시켜주었습니다.


정말 카카오톡처럼 구현해보고자 하니 어려운 부분이 두 개 있었습니다.

### 언제 상대방의 프로필과 이름을 띄우지?

상대방의 프로필과 이름이 항상 뜨면 안되고, 특정상황에만 떠야 했습니다.

- 처음 나한테 보냈다
- 이전 채팅이 상대방이 아니고 나다
- 날짜가 바뀌었다
 
3가지 경우의 수 + 내가 아닌 경우에는 상대방의 프로필과 사진을 띄워줬습니다.

### 타임스탬프가 분 단위로 같을때는 가장 밑에 것만 찍히게 하고싶어

`chat`객체를 만들때 `isDisplay:boolean`속성을 넣었고 최초에 만들어질때는 `true`를 갖게 했습니다. 그리고 새로운 채팅이 입력될 때마다 이전 채팅을 확인하고 이전채팅과 새로운 채팅의 발신자가 같은지, 날짜 및 시간이 같은지 확인하여 모두 일치한다면 이전 채팅의 `isDisplay`를 `false`로 바꿔주었습니다.

```ts
interface Chat {
  chatId: string;
  to: string;
  from: string;
  content: string;
  date: string;
  time: string;
  isDisplay: boolean;
  isRead: boolean;
}

interface ChattingRoom {
  id: string;
  chat: Chat[];
}
```

확장성을 고려해 채팅관련된 것을 하나의 객체배열 안에서 해결하고자 하여 `ChattingRoom`이라는 객체를 배열로 만들어 관리했고, 채팅이 갱신되면 `ChattingRoom`의 속성 중 하나인 `Chat[]`의 마지막 원소에 접근하고 새로운 원소를 추가해줘야 했습니다.

리액트는 상태의 불변성을 중요시 여기는데, 이렇게 중첩된 구조의 배열에 접근하여 값을 바꿔주어야 하다보니 `prev`의 값을 받아와 새로운 변수에 할당하고 이를 다시 넣어줘야하는 과정을 반복해야 했기 때문에 로직이 조금 복잡해졌습니다. 더 나은 방법을 고민해봤지만 떠오르지 않아 이렇게 해결했습니다.

날짜가 변경될 때도 날짜 변경선이 나오도록 구현했습니다.

카카오톡에서처럼 상단 navigation이 반투명하게 설정했습니다.

`useDoubleClick`훅을 만들어 상대방 채팅을 더블클릭하면 좋아요를 표시할 수 있도록 했습니다.

# Key Questions

## JavaScript를 사용할때에 비해 TypeScript를 사용할 때의 장점은 무엇인가요?

자동완성기능을 사용할 수 있고, 에러를 사전에 방지할 수 있습니다.

js는 유연한 언어인만큼 어떻게 작동할 지 모르는 부분들이 있습니다.

```javascript
[] + [] = ''
[] + {} = '[object object]'
{} + [] = 0
{} + {} = NaN
```

이렇게 원하지 않는 형태로 동작할 수 있습니다. 이처럼 예상치 못한 연산으로 에러가 발생하면 컴파일 에러를 띄워주지 않고 런타임에 에러를 띄웁니다. 타입스크립트는 이런 부분을 사전에 방지해줄 수 있습니다.

컴포넌트들이 많아지고 주고받는 props들이 생기거나 컴포넌트를 선언할 때 어떤 props가 필요한지 헷갈릴 수 있는데, 자동완성기능을 통해 어떤 속성들이 필요한지 알 수 있습니다.

타입을 너무 제한적으로 사용한다고 느낄 수도 있겠지만, 유니온타입이나 제네릭을 통해 타입을 확장성있게 사용할 수도 있습니다.

## 디자이너로부터 전달받은 피그마 링크 혹은, 피그마 캡처본

![캡쳐본](https://raw.githubusercontent.com/songess/CEOS_week3_reactMessenger/songess/src/assets/img/capture.png)

## 컴포넌트를 분리한 기준은 무엇인가요?

채팅 박스역할을 하는 코드들, 채팅의 상단바역할을 하는 코드들, 하단바 역할을 하는 코드들 처럼 기능에 따라 분류 했습니다.

```
src
 ┣ assets
 ┃ ┣ data
 ┃ ┃ ┗ chattingRoomData.json
 ┃ ┣ img
 ┃ ┃ ┗ ...
 ┃ ┣ svg
 ┃ ┃ ┣ ...
 ┣ hooks
 ┃ ┗ useDoubleClick.tsx
 ┣ pages
 ┃ ┗ ChattingRoomPage
 ┃ ┃ ┣ components
 ┃ ┃ ┃ ┣ ChatBottom.tsx
 ┃ ┃ ┃ ┣ ChatHead.tsx
 ┃ ┃ ┃ ┣ Chatting.tsx
 ┃ ┃ ┃ ┣ HeartBox.tsx
 ┃ ┃ ┃ ┣ Mychat.tsx
 ┃ ┃ ┃ ┣ OppoChangeChat.tsx
 ┃ ┃ ┃ ┗ OppoChat.tsx
 ┃ ┃ ┗ ChattingRoomPage.tsx
 ┣ recoil
 ┃ ┣ chatAtom.ts
 ┃ ┗ userAtom.ts
 ┣ styles
 ┃ ┣ GlobalStyle.ts
 ┃ ┗ theme.ts
 ┣ types
 ┃ ┗ type.ts
 ┣ App.tsx
 ┣ index.tsx
 ┣ svg.d.ts
 ┗ types.d.ts
```

페이지폴더를 만들고, 컴포넌트 폴더를 만들어서 페이지에서 사용하는 컴포넌트들을 만들어줬습니다. 기능별로 컴포넌트들을 구현했기 때문에 원하는 위치에서 컴포넌트를 호출해주고 css만 설정해주면 페이지를 만들 수 있었습니다.

## 디자인 시스템을 적용하면서 느낀 점은 무엇인가요?

혼자 만들다 보면 색상이나 텍스트스타일을 미리 지정해놓는게 아니기 때문에 한번 해보고 렌더링 해보면서 어떤게 제일 적합한지 여러번 시도해봐야 했습니다. 하지만 `figma`를 통해 디자인을 받아서 개발했기 때문에 그런 고민을 할 필요가 없었습니다.

또한 이미 정해진 규격(ex. body1, body2, body3, ...)이 있고 그것들을 사용해 디자인을 해주셨기 때문에, 미리 `theme`으로 규격을 저장해놓으면 개발하면서 호출하기만 하면 되니깐 굉장히 편했습니다.


## 디자이너와 소통하며 느낀점은 무엇인가요?

기능적인 부분을 위주로 생각하다보니 디자인적 요소를 가끔 잘못 만들 때가 있습니다. 버튼이 의도한게 아니거나, 간격이 이상하거나 하는 부분들을 놓쳐서 배포했는데 이런 부분들을 피드백 해주셔서 나는 개발하는 부분에 더 집중하면 돼서 좋다! 라고 느꼈습니다. 

또한 디자이너와 개발자의 시선이 다르기 때문에 원활한 소통을 통해 프로젝트를 끝내면 훨씬 더 멋진 결과물의 나올 거 같다고 생각했습니다.

songess committed:
chore: .prettierrc 파일 생성

songess committed:
chore: styled-components 설치

songess committed:
style: globalStyle 및 theme 설정

songess committed:
feat: svg, svg.d.ts 파일 생성

songess committed:
feat: 필요없는 파일 제거

songess committed:
chore: 절대경로 설정

songess committed:
feat: tsconfig.paths.json에 assets 절대 경로 추가

songess committed:
feat: Chathead 컴포넌트 구현

songess committed:
ChatBottom 컴포넌트 구현

songess committed:
fix: ChatHead margin 수정

songess committed:
chore: recoil 설치
feat: chatAtom 생성

songess committed:
feat: 이름 클릭 시 반전

songess committed:
feat: 메시지 출력함수 구현

songess committed:
fix: 메시지 출력함수 수정

songess committed:
feat: MyChat 구현

songess committed:
feat: OppoChat 구현

songess committed:
feat: OppoChangeChat 구현

songess committed:
feat: type.d.ts 파일 생성

songess committed:
feat: Chatting 컴포넌트 구현

songess committed:
feat: 채팅 추가기능 구현

songess committed:
feat: 날짜변경 구현

songess committed:
feat: 채팅창 overflow시 동작 구현

songess committed:
feat: 날짜가 바뀔때도 OppochangeChat 컴포넌트 호출

songess committed:
feat: localStorage를 통한 상태관리 구현

songess committed:
feat: 배포테스트

songess committed:
style: 이름과 말풍선 간격 조정

songess committed:
refactor: 전송 버튼 교체

songess committed:
feat: 더블클릭 커스텀훅 생성

songess committed:
style: 하트컴포넌트 스타일 추가

songess committed:
refactor: 윈도우에서 깨짐현상 수정
refactor: DUMMYDATA 수정

songess committed:
feat: readme.md 추가

Shunamo commented on src/styles/theme.ts:
디자이너님께서 명시해주신 색상명대로 따로 저장해서 사용하시니까 스타일 조정하시면서 편안했을 것 같아요! 저도 다음 주차 미션하면서 활용해서 써봐야겠어요..ㅎㅎㅎ 
theme으로 스타일 불러오시는거 보니까 진짜 편해보이네요!

Shunamo commented on src/pages/ChattingRoomPage/components/Chatting.tsx:
```diff
@@ -0,0 +1,146 @@
+      );
+    }
+    return messageComponent;
+  };
```
오.. 메세지 종류에 따라서 컴포넌트를 다르게 랜더링하는 것도 좋은 방식인 것 같아요! 
날짜 비교하는 로직은 함수로 분리시켜도 좋을 것 같아요~!!

Shunamo commented on src/pages/ChattingRoomPage/components/Chatting.tsx:
```diff
@@ -0,0 +1,146 @@
+      const month = chat.date.slice(5, 7);
+      const day = chat.date.slice(8, 10);
+      return (
+        <Fragment key={idx + 100}>
```
날짜랑 메세지 컴포넌트 간의 key값 충돌을 방지하시려고 매직넘버를 사용하신 것 같아요..!! 
근데 매직넘버를 사용하면 앞으로의 확장에 있어서 다른 충돌이 발생할 수도 있대요! 그래서 100을 더하는 것보다는 메세지의 고유 id를 정의해서 key값으로 사용하거나 다른 방법을 사용하는게 더 좋을 것 같아요! 
[매직넘버](https://devtist.tistory.com/20)

Shunamo commented on src/pages/ChattingRoomPage/components/Mychat.tsx:
```diff
@@ -0,0 +1,69 @@
+  display: flex;
+  align-items: flex-end;
+  padding: 10px;
+  background-color: ${theme.colors.white};
```
지워주기!

Shunamo commented on src/pages/ChattingRoomPage/components/Mychat.tsx:
```diff
@@ -0,0 +1,69 @@
+  font-size: ${theme.textStyles.caption2.font_size};
+  line-height: ${theme.textStyles.caption2.line_height};
+  font-weight: ${theme.textStyles.caption2.font_weight};
+`;
```
`MychatTime`이랑 `MychatIsRead`가 거의 같은 스타일인데, 이 둘을 합쳐서 조건부 랜더링으로 색상만 다르게 해줘도 좋을 것 같아요~!

Shunamo commented on src/pages/ChattingRoomPage/components/OppoChangeChat.tsx:
```diff
@@ -0,0 +1,111 @@
+  border-radius: 12px;
+  max-width: 240px;
+  word-wrap: break-word;
+  background-color: ${theme.colors.white};
```
지워주기!

Shunamo commented on src/pages/ChattingRoomPage/components/OppoChangeChat.tsx:
```diff
@@ -0,0 +1,111 @@
+  time: string;
+  isDisplay: boolean;
+  from: string;
+}
```
상대방 채팅이랑 내 채팅을 다른 컴포넌트로 분리해서 사용하는게 나중을 위해서 더 좋은 방법 같아요 :) 저는 이번에 분리를 제대로 못하고 개발을 진행한 것 같아서.. 아쉬웠는데 참고해서 리팩토링 해보려구요 👍

Shunamo commented on src/pages/ChattingRoomPage/components/OppoChat.tsx:
```diff
@@ -0,0 +1,69 @@
+  display: flex;
+  align-items: flex-end;
+  padding: 10px;
+  background-color: ${theme.colors.white};
```
지워주기!

Shunamo commented on src/recoil/chatAtom.ts:
```diff
@@ -0,0 +1,46 @@
+  time: string;
+  isDisplay: boolean;
+  isRead: boolean;
+}
```
읽음 표시까지...! 디테일 좋아요! 완성본이 기대돼요 ㅎㅎ

youdame replied:
Chatting과 chatAtom에서 Chat이라는 아예 같은 type 선언이 존재하는데 이런 경우에는 더더욱 types 폴더에서 관리하고 export - import 해서 사용하는 게 좋을 거 같습니다!

Shunamo commented on src/pages/ChattingRoomPage/ChattingRoomPage.tsx:
```diff
@@ -0,0 +1,26 @@
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  width: 375px;
```
이번 미션에 화면 크기가 고정값으로 주어져서 375*812로 진행하셨는데, 모바일로 접속하면 좌우에 공간이 남아서 분리된 느낌이 들더라구요. 
저는 그래서 데스크탑뷰에서만 `max-width`랑 `max-height`, `min-height` 속성으로 크기를 고정시켜줬구 
모바일에서는 뷰포트로 설정되도록 해놨어요! 이렇게 하면 실제 채팅방처럼 보여서 좋더라구요!

Shunamo commented on src/pages/ChattingRoomPage/components/ChatBottom.tsx:
```diff
@@ -0,0 +1,134 @@
+    return `${hour}:${minute}`;
+  };
+
+  const HandleSubmit = () => {
```
`handleSubmit`으로 이름변경해주기..!

Shunamo commented on src/pages/ChattingRoomPage/components/ChatHead.tsx:
```diff
@@ -0,0 +1,92 @@
+      </MenuBox>
+    </ChatHeaderWrapper>
+  );
+}
```
아이콘을 컴포넌트로 사용해서 구현하니까 가독성이 되게 좋은 것 같아요 저는 항상 컴포넌트마다 불러와서 사용했었는데.. 좋은 방법 배워갑니다 👍

Shunamo commented on src/pages/ChattingRoomPage/components/ChatHead.tsx:
```diff
@@ -0,0 +1,92 @@
+export default function ChatHead() {
+  const [userName, setUserName] = useRecoilState(UserState);
+  const displayName = userName === '송은수' ? '플래시' : '송은수';
+  const HandleNameToggle = () => {
```
handleNameToggle로 바꿔주기!!

Shunamo commented on src/pages/ChattingRoomPage/components/ChatHead.tsx:
```diff
@@ -0,0 +1,92 @@
+  height: 28px;
+  padding: 0 16px;
+  background-color: ${theme.colors.blue};
+  opacity: 0.8;
```
헤더에 투명도를 조절해서 위의 채팅이 보여지게 하신 것까지 카톡과 똑같네요! 예뻐요 💯

youdame replied:
저도 동의해요 ~~ 디테일 쩜

Shunamo commented on src/pages/ChattingRoomPage/components/ChatBottom.tsx:
```diff
@@ -0,0 +1,134 @@
+      setChattingData((prev) => {
+        const newChats = [...prev[0].chat, newChat];
+        return [{ ...prev[0], chat: newChats }, ...prev.slice(1)];
+      });
```
조건문으로 구현하니까 훨씬 이해하기 편하네요! 말씀해주신 것처럼 비교대상을 첫번째 채팅으로 해야 명확하게 시간대별로 메세지들을 분리시킬 수 있는 것 같아요 좋은 코드 배워갑니다!! 👍 👍 👍

youdame replied:
빈 입력값에 대한 예외 처리가 되어있지 않네요~
<img width="300" alt="스크린샷 2024-03-31 오후 12 28 19" src="https://github.com/CEOS-Developers/react-messenger-19th/assets/112458620/7520ff8d-9bd9-4b13-865c-fbfc929206fb">

Shunamo reviewed (commented):
은수님! 복잡한 로직들을 되게 간결하고 이해하기 쉽게 작성하셔서 리뷰하는데 편안했습니다..! 특히, 꼬리물기 기능을 구현하실 때, 중첩된 구조를 조건문으로 효율적으로 처리하는 방식이 저의 접근법과 달라 매우 인상 깊었습니다. 가독성 있는 코드를 작성하는 것의 중요성을 다시금 깨닫고 가네요..!! 많이 배워갑니다. 이번 주차도 정말 수고 많으셨습니다! 👍👍👍

youdame commented on src/pages/ChattingRoomPage/components/ChatHead.tsx:
```diff
@@ -0,0 +1,92 @@
+      <ChatTitle onClick={HandleNameToggle}>{displayName}</ChatTitle>
+      <MenuBox>
+        <SearchSvg />
+        <MenuSvg />
```
저는 클릭 가능한 svg는 일괄적으로 button 태그에 감싸주는데요! 그렇게 하면 이벤트 핸들러를 달기도 쉬워집니다 ㅎㅎ 그리고 globalStyle에서

```
button {
  cursor:pointer
}
```
를 해주면 하나하나 cursor : pointer를 줄 필요도 없어져서요!

youdame commented on src/pages/ChattingRoomPage/components/ChatHead.tsx:
```diff
@@ -0,0 +1,92 @@
+
+const DUMMYLEFTCOUNT = 3;
+
+export default function ChatHead() {
+  const [userName, setUserName] = useRecoilState(UserState);
+  const displayName = userName === '송은수' ? '플래시' : '송은수';
```
이후에는 어떤 채팅방을 선택하냐에 따라 유저의 이름이 달라질테니 그 방향성을 생각해서 리팩토링 해보면 좋을 거 같아요~

youdame commented on src/pages/ChattingRoomPage/components/Chatting.tsx:
```diff
@@ -0,0 +1,146 @@
+import OppoChangeChat from './OppoChangeChat';
+import Oppochat from './OppoChat';
+
+interface Chat {
+  chatId: string;
+  to: string;
+  from: string;
+  content: string;
+  date: string;
+  time: string;
+  isDisplay: boolean;
+  isRead: boolean;
+}
```
저는 일괄적으로 prop에 사용할 타입이 아니라면, types 폴더에 페이지별 타입을 저장해놓고 import 해서 사용하는 편이에요!

youdame commented on src/pages/ChattingRoomPage/components/Chatting.tsx:
저라면 날짜 표시 로직을 별도의 컴포넌트로 분리하고, showMessage에서 리턴하는 컴포넌트의 각각의 프롭 이름이 모두 같으니 이런 식으로 작성했을 거 같아요! 코드를 온전히 이해하고 작성한 건 아니어서 틀린 로직이 있을 수도 있으니 참고만 해주세요:)

```ts

  const MessageDisplay = ({ chat, idx }: { chat: Chat; idx: number }) => {
    const isMyMessage = chat.from === userName;
    const isNewSenderOrDate =
      idx === 0 ||
      chat.from !== chattingData?.chat[idx - 1].from ||
      chat.date !== chattingData?.chat[idx - 1].date;

    const Component = isMyMessage
      ? Mychat
      : isNewSenderOrDate
      ? OppoChangeChat
      : Oppochat;

    return (
      <Fragment key={chat.chatId}>
        <DateDisplay chat={chat} idx={idx} />
        <Component
          content={chat.content}
          isDisplay={chat.isDisplay}
          isRead={chat.isRead}
          time={chat.time}
          from={!isMyMessage && chat.from}
        />
      </Fragment>
    );
  };

  return (
    <ChatWrapper ref={chatRef}>
      <SpaceBox />
      {chattingData?.chat.map((chat, idx) => (
        <MessageDisplay key={idx} chat={chat} idx={idx} />
      ))}
    </ChatWrapper>
  );
```

youdame commented on src/pages/ChattingRoomPage/components/Chatting.tsx:
```diff
@@ -0,0 +1,146 @@
+  line-height: ${theme.textStyles.caption4.line_height};
+  font-weight: ${theme.textStyles.caption4.font_weight};
+  text-align: center;
+  &::before {
+    display: inline-block;
+    width: calc(40% - 25px);
+    height: 1px;
+    margin: 8px 10px 8px 0;
+    background-color: ${theme.colors.gray_2};
+    vertical-align: top;
+    content: '';
+  }
+
+  &::after {
+    display: inline-block;
+    width: calc(40% - 25px);
+    height: 1px;
+    margin: 8px 0 8px 10px;
+    background-color: ${theme.colors.gray_2};
+    vertical-align: top;
+    content: '';
+  }
+`;
```
가상 선택자를 잘 활용하시는 군요 ㅎㅎ 저는 before, after는 한 번도 사용해본 적 없는데 배워갑니다!

youdame commented on src/pages/ChattingRoomPage/components/ChatBottom.tsx:
```diff
@@ -0,0 +1,134 @@
+  const [inputValue, setInputValue] = useState('');
+  const userName = useRecoilValue(UserState);
+  const [chattingData, setChattingData] = useRecoilState(chatDataState);
+
+  const formatDate = (date: Date) => {
+    const year = date.getFullYear().toString();
+    let month = (date.getMonth() + 1).toString().padStart(2, '0');
+    let day = date.getDate().toString().padStart(2, '0');
+    return `${year}-${month}-${day}`;
+  };
+
+  const formatTime = (date: Date) => {
+    let hour = date.getHours().toString().padStart(2, '0');
+    let minute = date.getMinutes().toString().padStart(2, '0');
+    return `${hour}:${minute}`;
+  };
+
```
이런 로직들이 컴포넌트 안에 있게 되면 리렌더링이 될 때마다 함수가 새로 만들어져요. 그래서 보통은 utils 폴더 안에 넣어 정리해준답니다. 또한 const와 let 선택자를 혼용해서 사용하고 계신데, 저는 주로 const를 사용해서 의도치 않는 값의 재할당을 막아주는 편입니다!

youdame commented on src/pages/ChattingRoomPage/components/ChatBottom.tsx:
```diff
@@ -0,0 +1,134 @@
+          ...prev[0].chat[lastChatIndex],
+          isDisplay: false,
+        };
+        lastChat.isDisplay = false;
```
61번째 코드는 불필요해보이는데 맞을까요~?

youdame commented on src/pages/ChattingRoomPage/components/ChatBottom.tsx:
```diff
@@ -0,0 +1,134 @@
+          onChange={(e) => {
+            setInputValue(e.target.value);
+          }}
+          onKeyPress={(e) => {
+            if (e.key === 'Enter') {
+              HandleSubmit();
+            }
```
onkeyPress 속성이 deprecated 되었다고 하니 다른 이벤트를 사용해보는 게 좋을 거 같아요! 
[onkeypress](https://www.w3schools.com/jsref/event_onkeypress.asp)

그리고 form 태그 안에 input과 버튼을 두면 엔터와 아이콘 클릭 이벤트를 굳이 분리할 필요 없이 처리가 form 태그의 onSubmit에 handleSubmit 함수를 연결해준다면 한 번에 처리가 됩니다~

youdame commented on src/pages/ChattingRoomPage/ChattingRoomPage.tsx:
```diff
@@ -0,0 +1,26 @@
+  align-items: center;
+  justify-content: center;
+  width: 375px;
+  height: 100vh;
```
height 값은 고정 안해주신 건가요~?

youdame reviewed (commented):
은수님 고생 많으셨어요! 몇 가지 코멘트 남겨뒀으니 참고 부탁드려요 ㅎㅎ 

> 375x812 맞춤으로 코드를 작성할 것이라 rem단위를 사용하지 않고 px을 사용했습니다.

라는 게 어떤 의미일까요?

rem은 사용자의 브라우저 설정이나 기기의 디스플레이 특성에 따라 폰트 크기가 조정될 수 있기에 사용한다고 알고 있어서 모바일 반응형으로 고정되어있을 땐 px을 사용하는 게 더 나은지는 잘 모르겠어서요..!!


제가 리코일을 사용해본 적이 없어서 더 나은 코드리뷰를 드리지 못한 점이 아쉽네요ㅠㅠ 아쉬운 점 위주로 적어봤는데, 다음에는 칭찬도 마구마구 적어보겠씁니다~~!!!!!!

songess committed:
style: 반응형 구현

songess committed:
fix: 스타일 수정 및 반응형 구현
- position: fixed -> absolute로 변경
- status bar도 반응형으로 구현
컴포넌트 구조 변경

songess committed:
fix: 모바일 화면구성을 위해 vh->dvh로 변경

songess committed:
fix: margin-bottom 조정

songess committed:
fix: ChatBottom 스타일 수정 및 position 변경
- input과 '+'간격 조정
- input과 '+'가운데정렬
- input 스트로크 추가
- position: absolute -> fixed 로 변경

songess committed:
fix: Chatting 최하단 margin 수정

songess committed:
fix: 모바일환경에서 'pretendard'폰트 적용되도록 수정
